Alexandr Darvin

fruitella44 opened the pull request:
Не получается записать данные в файл. После ввода данных пользователя, метод создает каждый раз новые файлы

fruitella44 committed:
changed scanner

fruitella44 committed:
cipher to the text.txt

fruitella44 committed:
cipher to the text.txt

fruitella44 commented:
Создал отдельный файл. Считываем с консоли, что пишет пользователь и заносим зашифрованные данные в файл text.txt
Это пока мой потолок, дальше пока не знаю как делать.

fruitella44 committed:
added getFilePath

fruitella44 committed:
refactoring encode and decode. Now encode/decode works with Writter into file

fruitella44 committed:
updated readFile, encode and decode

fruitella44 committed:
added option to choose a key

fruitella44 committed:
added option to choose a key

fruitella44 committed:
brute force doesn't work

fruitella44 committed:
trying to fix bruteForce

fruitella44 committed:
brute force doesn't work

fruitella44 committed:
brute force doesn't work

fruitella44 committed:
need help with brute force

fruitella44 committed:
Added classes: Encode, Decode, BruteForce
BruteForce doesn't work

fruitella44 committed:
Fixed BruteForce, changed readmi

akupreychik commented on src/ua/com/javarush/darvin/module1/BruteForce.java:
```diff
@@ -0,0 +1,58 @@
+        StringBuilder inputText = new StringBuilder(Main.readFile(line).toString().toLowerCase());
+        StringBuilder outputText = new StringBuilder();
+
+        while (key > 0 && key < 43) {
```
волшебные цифры! выноси все в константы!

akupreychik commented on src/ua/com/javarush/darvin/module1/Main.java:
```diff
@@ -0,0 +1,125 @@
+
+    }
+
+    private int scannerInt() {
```
название методов очень не подхоядщие - у тебя старт программы в этой точке

akupreychik commented on src/ua/com/javarush/darvin/module1/Main.java:
```diff
@@ -0,0 +1,125 @@
+        }
+
+       return false;
+    }
```
все методы возвращают boolean. но эта переменная ни где не используется

akupreychik commented on src/ua/com/javarush/darvin/module1/Main.java:
```diff
@@ -0,0 +1,125 @@
+        }
+
+        return false;
+    }
```
методы возвращают boolean. но эта переменная ни где не используется

akupreychik commented on src/ua/com/javarush/darvin/module1/Main.java:
```diff
@@ -0,0 +1,125 @@
+        return false;
+    }
+
+    private static boolean scannerBruteForce() {
+        BruteForce brute = new BruteForce();
+        Scanner console = new Scanner(System.in);
+        String line = console.nextLine();
+
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter("E:\\Coding\\Java\\javarush-cryptoanalyser\\inputText.txt"))) {
+
+            bufferedWriter.append(brute.bruteForce(line, 1));
+            bufferedWriter.flush();
+            return true;
+        }catch (IOException exception) {
+            System.out.println("Не удалось прочитать текст " + exception);
+        }
+
+        return false;
+    }
```
методы возвращают boolean. но эта переменная ни где не используется

akupreychik commented on src/ua/com/javarush/darvin/module1/Main.java:
```diff
@@ -0,0 +1,125 @@
+        Scanner console = new Scanner(System.in);
+        String line = console.nextLine();
+
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter("E:\\Coding\\Java\\javarush-cryptoanalyser\\inputText.txt"))) {
```
а где эти файлы? а что будет если я запущу программу с мака или линукса?

akupreychik commented on src/ua/com/javarush/darvin/module1/Main.java:
```diff
@@ -0,0 +1,125 @@
+
+    private static boolean scannerEncode() {
+        Encode encrypt = new Encode();
+        Scanner console = new Scanner(System.in);
```
сканнер можно вынести на уровень общения с пользователем. чтобы не создавать его каждый раз

akupreychik commented on src/ua/com/javarush/darvin/module1/Main.java:
```diff
@@ -0,0 +1,125 @@
+        } catch (InputMismatchException exception) {
+            System.out.println("Требуется ввести число! " + exception);
+        }
+        return input;
```
твоя программа выполнится только один раз. оберни в цикл и так будет удобнее; не надо будет запускать ее каждый раз

akupreychik commented on src/ua/com/javarush/darvin/module1/Main.java:
```diff
@@ -0,0 +1,125 @@
+    private int scannerInt() {
+        Scanner console = new Scanner(System.in);
+
+        System.out.println("Запуск программы!\n___________________________");
+        System.out.println("Выберите действие:");
+        System.out.println("1 - шифровка текста");
+        System.out.println("2 - расшифровка текста");
+        System.out.println("3 - расшифровка с помощью brute force");
+        System.out.println("0 - Выход\n___________________________");
```
лучше использовать блоки строк """ много текста """

akupreychik commented on src/ua/com/javarush/darvin/module1/Main.java:
```diff
@@ -0,0 +1,125 @@
+
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter("E:\\Coding\\Java\\javarush-cryptoanalyser\\inputText.txt"))) {
+
+            System.out.println("___________________________\nВведите номер ключа:");
```
вынеси __________________ в отдельную константу

akupreychik commented on src/ua/com/javarush/darvin/module1/BruteForce.java:
```diff
@@ -0,0 +1,58 @@
+                outputText.append(newChars);
+            }
+
+            String[] splitString = outputText.toString().split("[\\p{P} \\t\\n\\r]");
```
regex лучше тоже вынести в константу

akupreychik reviewed (commented):
хороший проект! молодец!

fruitella44 committed:
Refactoring scannerInt, added while (true).
Regex and messages initialization into constants.
Changed Path, now it generates a new file after decoding

fruitella44 committed:
added META-INF into gitignore

AndryLat commented on src/ua/com/javarush/darvin/module1/BruteForce.java:
```diff
@@ -0,0 +1,57 @@
+            "какой", "после", "их", "работа", "без", "самый", "потом", "надо", "хотеть", "ли", "слово", "идти",
+            "большой", "должен", "место", "иметь", "ничто"};
+
+    public static final Pattern PATTERN = Pattern.compile("[\\p{P} \\t\\n\\r]");
```
Непонятно что именно за паттерн.

AndryLat commented on src/ua/com/javarush/darvin/module1/BruteForce.java:
```diff
@@ -0,0 +1,57 @@
+
+            for (int i = 0; i < inputText.length(); i++) {
+                int index = ALPHABET.indexOf(inputText.charAt(i));
+                int shiftIndexBack = (index - key) % 43;
```
особенно проблема, что 43 - это длина алфавита, а завтра добавим новую букву и везде надо бегать по коду ии скать где еще исправить.

AndryLat commented on src/ua/com/javarush/darvin/module1/BruteForce.java:
```diff
@@ -0,0 +1,57 @@
+                outputText.append(decoding);
+            }
+
+            String[] splitString = outputText.toString().split(PATTERN.toString());
```
можно назвать words, это будет более емким

AndryLat commented on src/ua/com/javarush/darvin/module1/Decode.java:
```diff
@@ -0,0 +1,26 @@
+package ua.com.javarush.darvin.module1;
+
+public class Decode {
+    private static final String ALPHABET  = "абвгдеёжзийклмнопрстуфхцчшщъыьэюя.,«»\"\\:!? ";
```
зачем дублировать алфавит везде?

fruitella44 replied:
> зачем дублировать алфавит везде?

У меня сперва был encode, decode, bruteforce в одном классе. Антону не понравилось, сказал сделай encode, decode и bruteforce в отдельных классах. При этом ты говорил присвоить алфавиту модификатор private, но если private я не могу обратиться к этой константе из другого класса. Какая то безвыходная ситуация...

AndryLat commented on src/ua/com/javarush/darvin/module1/Decode.java:
```diff
@@ -0,0 +1,26 @@
+    private static final String ALPHABET  = "абвгдеёжзийклмнопрстуфхцчшщъыьэюя.,«»\"\\:!? ";
+
+    public static StringBuilder decode(String line, int key) {
+        StringBuilder inputText = new StringBuilder(Main.readFile(line).toString().toLowerCase());
```
Main вызывает Decode, а Decode вызывает Main. это сильносвязанные классы, такое надо избегать. иначе при хоть каких-то изменениях все посыпется.

AndryLat commented on src/ua/com/javarush/darvin/module1/Decode.java:
```diff
@@ -0,0 +1,26 @@
+
+        for (int i = 0; i < inputText.length(); i++) {
+            int index = ALPHABET.indexOf(inputText.charAt(i));
+            int shiftIndexBack = (index - key) % 43;
```
также код очень похож на энкод.

AndryLat commented on src/ua/com/javarush/darvin/module1/Main.java:
```diff
@@ -0,0 +1,132 @@
+                try {
+                    int input = Integer.parseInt(command.nextLine());
+
+                    if (input == 1) {
```
Что такое 1, 2, 3?

AndryLat replied:
Нужно вынести в именованую константу

fruitella44 replied:
Так я вынес, сообщение наверху в Main классе, я не понимаю

AndryLat commented on src/ua/com/javarush/darvin/module1/Main.java:
```diff
@@ -0,0 +1,132 @@
+        Scanner console = new Scanner(System.in);
+        String line = console.nextLine();
+
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter("inputText.txt"))) {
```
строка inputText.txt дублируется, лучше вынести в константу.

fruitella44 committed:
added ALPHABET_LENGTH

fruitella44 committed:
changed Alphabet.length();

fruitella44 committed:
Added class Alphabet. Encode/Decode/BruteForce now extends to Alphabet
Added class ReadAndWrite. Now read and write text into files works in class ReadAndWrite
Changed value of alphabet. It used to be 43, now it gets ALPHABET.length(); for case if we would add some letters else in our Alphabet.
Everying works correctly, i've done tests.

fruitella44 committed:
changed readmi

fruitella44 committed:
changed readmi
